db/src/tx_observer.rs

```diff
@@ -104,6 +99,7 @@ pub struct AsyncBatchExecutor {
 
 impl Command for AsyncBatchExecutor {
     fn execute(&self) {
+        // need to clone to move to a new thread.
         let command_queue = self.commands.clone();
         thread::spawn (move ||{
```

Comment:
Nit: space before `{`.

Reply:
If you spawn a new thread each time, it's possible for the first report in your second `InProgress` to be processed by observers before the last report in your first `InProgress`, no? This threading model will need to be revisited a bit later: you really want a thread to be spawned some time after the first observer is registered and before the first commit, to live until the last observer is unregistered, and to consume commands from a _shared_ queue.